core/src/term/mod.rs

```diff
@@ -681,6 +681,20 @@ impl fmt::Display for MergePriority {
     }
 }
 
+impl Serialize for MergePriority {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            Self::Bottom => serializer.serialize_unit_variant("priority", 0, "default"),
+            Self::Neutral => serializer.serialize_unit_variant("priority", 3, "neutral"),
+            Self::Numeral(n) => serializer.serialize_newtype_variant("priority", 2, "numeral", &n),
+            Self::Top => serializer.serialize_unit_variant("priority", 3, "force"),
+        }
+    }
+}
```

Comment:
This looks pretty mechanic - can't `serde` derive it automatically with `#[derive(Serialiaze)]` on `MergePriority` ?